### ISSUE TITLE
[BUG] NullReferenceException when accessing IStorageAccount.Region

### ISSUE THREAD

**Issue by maxim-gavrilov-veeam** [Azure/azure-libraries-for-net#681]
**Describe the bug**
You get NullReferenceException when accessing IStorageAccount.Region.

***Exception or Stack Trace***
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.Region.Create(String name)
   at [My Code]
```
(this is very short because most of the methods have been inlined).

**To Reproduce**
Get StorageAccount in any way and query Region property.

***Code Snippet***
```
var storage = await azure.StorageAccounts.GetByIdAsync(id, ct);
var region = storage.Region;
```

**Expected behavior**
I expect to get a Region instance.

**Setup (please complete the following information):**
 - OS: Windows
 - IDE : VS 2017
 - Version of the Library used: 1.22

**Additional context**
This is probably happening because *StorageAccountInner* inherits *TrackedResourceInner*, which has its *Location* property "overridden" via *new* keyword.
*ResourceBase.Region*, however, uses *Location* property from *Resource*.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by praries880** [comment]
Thank you for opening this issue! We are routing it to the appropriate team for follow up.

**Comment by maximrub** [comment]
any news about this bug?
we cannot upgrade to the new version because of this, is there a workaround we can use?

**Comment by maxim-gavrilov-veeam** [comment]
@maximrub, you can use `Region.Create(storageAccount.Inner.Location)` as a workaround.

**Comment by ChenTanyi** [comment]
Hi @maxim-gavrilov-veeam  @maximrub ,

I've tested it from 1.22.0 to 1.25.0, and found it was fixed in 1.24.0. You could upgrade the library to 1.24.0 or later to fix the issue.

Thanks for this issue. Now I'll close it due to it was fixed.

If there is still something wrong about it, feel free to reopen it. 

![image](https://user-images.githubusercontent.com/15213781/64749544-8ce9b980-d548-11e9-8e9a-f5170fe4db32.png)

**Comment by yaohaizh** [comment]
Fixed on 1.24.0
